### DISCUSSION TITLE
User Profile Metadata is sent unnecessarily by admin console when users updated

### DISCUSSION THREAD

**Discussion by jbman** [keycloak/keycloak#25148] [Ideas]
With Keycloak 22, the Admin Console always requests user profile metadata with the GET request for a user and sends it back on update (PUT request):
```
{
    "id": ...,
    ...
    "userProfileMetadata": {... }
}
```

Probably the endpoint shouldn't do anything with the data so that it could be removed by the console application before the PUT.

In our case, user update got blocked by the application firewall, because the metadata contained validator patterns which were considered suspicious. While we already configured firewall exceptions for defining the profile, now user updates started failing.

It would be good to reduce the unnecessary payload when the user is updated.

**Comment by jonkoops** [comment]
Feel free to submit this as a bug

**Reply by jbman** [reply]
Thanks, added https://github.com/keycloak/keycloak/issues/25463
